tests/conftest.py

```diff
@@ -128,3 +131,8 @@ def reload_directory_structure(tmp_path_factory: pytest.TempPathFactory):
     ext_file.touch()
 
     yield root
+
+
+@pytest.fixture(scope="function")
```

Comment:
using the log_config fixture on a function scope to ensure we get a clean one each test and that when we mess up with caplog pytest internals we dont break things